internal/command/user_v2_human.go

```diff
@@ -302,6 +319,36 @@ func (c *Commands) ChangeUserHuman(ctx context.Context, human *ChangeHuman, alg
 		}
 	}
 
+	for _, md := range human.Metadata {
+		cmd, err := c.setUserMetadata(ctx, userAgg, md)
+		if err != nil {
+			return err
+		}
+
+		cmds = append(cmds, cmd)
+	}
+
+	for _, mdKey := range human.MetadataKeysToRemove {
+		cmd, err := c.removeUserMetadata(ctx, userAgg, mdKey)
+		if err != nil {
+			return err
+		}
+
+		cmds = append(cmds, cmd)
+	}
+
+	if human.State != nil {
```

Comment:
Can you combine the logic with `DeactivateUserV2`, `ReactivateUserV2`? So that we have no logic changes?

Reply:
The logic is actually quite different. For instance, the checks in the `DeactivateUserV2` function and there equivalent in `ChangeUserHuman`:

- `isUserStateExists`: This check is already performed early in the `ChangeUserHuman` function.
- `isUserStateInitial`: This check is handled in the `ChangeUserHuman` function.
- `isUserStateInactive`: The `ChangeUserHuman` function generally ignores updates that are not needed or unchanged values.

Out of these three checks, only one is handled equally. I'm not sure how exactly the expected solution would look like. Could you please provide an example of how these methods/code segments should be combined?

Reply:
What I mean is exactly the `isUserStateInactive`, as with this check we guarantee that you can only reactivate a deactivated user (and deactivate only an active user).
The `isUserStateInactive` is not meant to ignore unneeded changes, but only enable you to change the state of a user if they are in a specific state.
 
 To be on the save side that the same logic is used in the different endpoints.

Reply:
Adjusted as discussed:
* only allow switching to inactive if the current state of the user is active
* only allow switching to active if the current state of the user is inactive
* if the target state is the same as the current state no command is pushed
* all other state transitions result in an error being returned
* the state should be evaluated using the `isUserState*` functions